ui-core/src/components/inputs/TimePicker.tsx

```diff
@@ -47,23 +64,56 @@ const TimePicker = ({ onTimeChange, hours = 0, minutes = 0, ...otherProps }: Tim
       newMinutes = 59;
       newHours = newHours === 0 ? 23 : newHours - 1;
     }
-    onTimeChange({ hours: newHours, minutes: newMinutes });
+    onTimeChange({ hours: newHours, minutes: newMinutes, seconds });
+  };
+
+  const incrementSeconds = (increment: number) => {
+    let newSeconds = seconds + increment;
+    let newMinutes = minutes;
+    let newHours = hours;
+
+    if (newSeconds >= 60) {
+      newSeconds = 0;
+      newMinutes += 1;
+
+      if (newMinutes >= 60) {
+        newMinutes = 0;
+        newHours = newHours === 23 ? 0 : newHours + 1;
```

Comment:
```suggestion
        newHours = (newHours + 1) % 24;
```
Shorter but maybe less obvious ?

Reply:
no, it’s better. done